pkg/scene/cluster_list.go

```diff
@@ -0,0 +1,69 @@
+package scene
+
+import "github.com/reearth/reearth-backend/pkg/id"
+
+type ClusterList struct {
+	clusters []*Cluster
+}
+
+func NewClusterList() *ClusterList {
+	return &ClusterList{clusters: []*Cluster{}}
+}
+
+func NewClusterListFrom(clusters []*Cluster) *ClusterList {
+	return &ClusterList{clusters: clusters}
+}
+
+func (tl *ClusterList) Clusters() []*Cluster {
+	if tl == nil || tl.clusters == nil {
+		return nil
+	}
+	return append([]*Cluster{}, tl.clusters...)
+}
+
+func (tl *ClusterList) Has(tid id.ClusterID) bool {
+	if tl == nil || tl.clusters == nil {
+		return false
+	}
+	for _, cluster := range tl.clusters {
+		if cluster.ID() == tid {
+			return true
+		}
+	}
+	return false
+}
+
+func (tl *ClusterList) Add(clusters ...*Cluster) {
+	if tl == nil || tl.clusters == nil {
+		return
+	}
+	tl.clusters = append(tl.clusters, clusters...)
+}
+
+func (tl *ClusterList) Update(cid id.ClusterID, name string, pid id.PropertyID) {
+	if tl == nil || tl.clusters == nil || !tl.Has(cid) {
+		return
+	}
+
+	for _, c := range tl.clusters {
+		if c.ID() == cid {
+			c.Rename(name)
+			c.UpdateProperty(pid)
+		}
+	}
+}
```

Comment:
```suggestion
func (tl *ClusterList) Get(cid id.ClusterID) *Cluster {
	if tl == nil {
		return
	}

	for _, c := range tl.clusters {
		if c.ID() == cid {
			return c
		}
	}
	return nil
}
```

Cluster internals are not the responsibility of ClusterList, which only manages the list. ClusterList only manages multiple clusters, otherwise the code of ClusterList will get bloated.

```go
list.Get(cid).Rename(name)
```